gc.c

```diff
@@ -1884,6 +1884,52 @@ rb_gc_initial_stress_set(VALUE flag)
     initial_stress = flag;
 }
 
+static void * Alloc_GC_impl(void);
+
+#if USE_SHARED_GC
+# include "dln.h"
+# define Alloc_GC rb_gc_functions->init
+
+void
+ruby_external_gc_init()
+{
+    rb_gc_function_map_t *map = malloc(sizeof(rb_gc_function_map_t));
+    rb_gc_functions = map;
+
+    char *gc_so_path = getenv("RUBY_GC_LIBRARY_PATH");
+    if (!gc_so_path) {
+        map->init = Alloc_GC_impl;
+        return;
+    }
+
+    void *h = dln_open(gc_so_path);
+    if (!h) {
+        rb_bug(
+            "ruby_external_gc_init: Shared library %s cannot be opened.\n",
```

Comment:
`rb_bug` appends a newline.
```suggestion
            "ruby_external_gc_init: Shared library %s cannot be opened.",
```